### ISSUE TITLE
force redirects to https causes infinite loop

### ISSUE THREAD

**Issue by lilmike** [cydrobolt/polr#240]
Hi,
I'm using something like the following to redirect all http traffic to https:

server {
listen 80;
listen [::]:80;
server_name xxx;
return 301 https://$server_name$request_uri;
}

This, however, doesn't work. I always get as many redirects as my browser can handle, until it finally gives up. I've set the scheme to https:// in the setup, but still no go. When I include ports 80 and 443 in the same server block without redirecting, I can use both http and https. This is using nginx 1.11.4.
-Michael.

**Comment by cydrobolt** [comment]
Can I see your port 443 nginx config block?

**Comment by lilmike** [comment]
Hi,
Here is the entire nginx config. I've currently commented out the http only block because that's what makes it infinitely redirect.

#server {
#listen 80;
#listen [::]:80;
#    server_name 4mt.me www.4mt.me;
#return 301 https://$server_name$request_uri;
#}

server {
    access_log off;
    error_log /etc/nginx/logs/error.log;
listen 80;
listen [::]:80;
    listen 443 ssl;
listen [::]:443 ssl;
    server_name 4mt.me www.4mt.me;
            ssl_certificate /etc/letsencrypt/live/4mt.me/fullchain.pem;
            ssl_certificate_key /etc/letsencrypt/live/4mt.me/privkey.pem;
    index index.php index.html;

    root /storage/websites/4mt.me/polr/public;
location / {
try_files $uri $uri/ index.php?$is_args$args /index.php?$is_args$args;
}

location ~ \.php$ {
try_files $uri =404;
        fastcgi_split_path_info ^(.+\.php)(/.+)$;
include /etc/nginx/fastcgi_params;
        fastcgi_split_path_info ^(.+\.php)(/.+)$;
# fastcgi_pass unix:/run/php-fpm/php-fpm.sock; 
fastcgi_pass 127.0.0.1:9000;
#fastcgi_pass unix:/var/run/hhvm/hhvm.sock;
fastcgi_index index.php;
# fastcgi_param SCRIPT_FILENAME /scripts$fastcgi_script_name;
        fastcgi_param SCRIPT_FILENAME $document_root$fastcgi_script_name;
        fastcgi_param QUERY_STRING    $query_string;
#include fastcgi_params; 
        fastcgi_intercept_errors on;
        fastcgi_ignore_client_abort off;
        fastcgi_connect_timeout 10;
        fastcgi_send_timeout 10;
        fastcgi_read_timeout 30;
        fastcgi_buffer_size 128k;
        fastcgi_buffers 4 256k;
        fastcgi_busy_buffers_size 256k;
        fastcgi_temp_file_write_size 256k;

} 
}


-Michael.

**Comment by cydrobolt** [comment]
Have you tried removing `listen 80` from the SSL block, and changing it to only listen to 443? Then, you can have port 80 return a redirect to https://

**Comment by lilmike** [comment]
Hi,
Yes, actually, that's what I tried first. I tried to make it clear, but obviously it wasn't clear enough haha. The commented block on top is the block I had before, with no listen 80 or listen [::]:80 in the https block. However, when that block is active, it infinitely redirects. If I remove that block from the picture (I.E. comment it out like it is in the snippet I sent) and put listen 80 in the https block, it will work fine on both http or https, but obviously won't have the redirect.

Tl;dr: If I uncomment the http only block, and make the second one https only, I get the infinite redirect.
-Michael.

**Comment by BrainStone** [comment]
Did you change the protocol in the .env file to https?

`APP_PROTOCOL=https://`

**Comment by lilmike** [comment]
Yes, I changed the protocol to https.
Turns out, this was a problem with my cloudflare (that's what my domains are on) setup, where it was using port 443 (ssl encrypted on its end) to my port 80, (unencrypted). However, this made nginx think it was on http, so it would try to redirect, which of course cloudflare would just send it back to port 80. I fixed it so it actually connects via 443 (encrypted on my end) for ssl connections. Closing.
-Michael.

**Comment by cydrobolt** [comment]
Cool, glad you figured it out!
